src/firefly/js/externalSource/sprintf.js

```diff
@@ -73,7 +73,7 @@ function sprintf_format(parse_tree, argv) {
                 arg = arg();
             }
 
-            if (re.numeric_arg.test(ph.type) && (typeof arg !== 'number' && isNaN(arg))) {
+            if (re.numeric_arg.test(ph.type) && (!['bigint','number'].includes(typeof arg) && isNaN(arg))) {
```

Comment:
this might be more readable if you broke it out to a function.
```js
const isNumeric= (arg) => ['bigint','number'].includes(typeof arg)
```
later
```js
if (re.numeric_arg.test(ph.type) && (!isNumeric(arg) && isNaN(arg))) {
```

Reply:
I just did some testing at the console. Using a `reviver` function does not seem to work since the value is already converted when it is called. I am really surprised. I guess using the json replacement is fine thought I am still concerned about performance.